azurelinuxagent/common/future.py

```diff
@@ -24,3 +28,24 @@
 
 else:
     raise ImportError("Unknown python version: {0}".format(sys.version_info))
+
+
+def get_linux_distribution(get_full_name, supported_dists):
+    """Abstract platform.linux_distribution() call which is deprecated as of
+       Python 3.5"""
+    if float(sys.version[:3]) >= 3.5:
```

Comment:
This works. We also have the option of defining this method twice, one for `< 3.7` and one for `>= 3.7`. Depending on how the Python versions and deprecations change over time, one or the other may end up being more maintainable.

Reply:
As mentioned 3.7 does not work as check in SUSE, I am game to implement 2 methods or just on get_linux_distribution() and then have that be a redirector to _get_linux_distribution_py2() and _get_linux_distribution_py35p()

Reply:
I am good with either, your call.